scripts/dev/newReleaseBranch.py

```diff
@@ -0,0 +1,549 @@
+#######################################
+# This script creates HotFix branches for repo, share and packaging projects, and updates the master branches ready for the next SP/major release.
+# To use this script you need to install lxml:
+# 'pip install lxml'
+# Run the script without passing any arguments or with -h/--help argument to get usage information.
+# See below script behaviour explained.
+#######################################
+# Create a HotFix branches for the released version (for X.Y.Z release it will be X.Y.N eg., create 23.2.N for 23.2.0 release)
+# 1. acs-packaging:
+# - set RELEASE_VERSION to X.Y.Z+1, DEVELOPMENT_VERSION to X.Y.Z+1-SNAPSHOT in master_release.yml
```

Comment:
I think `Z` will always be `0` when calling this? I guess the DEVELOPMENT_VERSION will only be used when making the HF release and so should be higher than the RELEASE_VERSION.
```suggestion
# - set RELEASE_VERSION to X.Y.1, DEVELOPMENT_VERSION to X.Y.2-SNAPSHOT in master_release.yml
```

Reply:
Good point about `Z`.
However, script sets `DEVELOPMENT_VERSION` to 'Z+1' (which effectively is `1` instead of suggested `Z+2` = `2`).
At first I was setting that to `2` but after having looked at what happened historically I was confused and started setting that to `1`.
See following commits and let me know what you think:
https://github.com/Alfresco/acs-packaging/commit/b60e365d16018ae2b123c5bf015f337a80e5f3e9 (SP branch, dev version # same as release version #)
https://github.com/Alfresco/acs-packaging/commit/739bf7fd2d22efb6571f6c507728448ce0d37dc5 (HF branch, dev version # greater by `1` from release version #)
https://github.com/Alfresco/acs-packaging/commit/fdf4f38aa28c031c481ec949ef81d9474176f2f0 (master branch, dev version # same as release version #)
I'm not sure if it was historically correct, so  let me know what you think should be done here.

Reply:
So summarising those three commits - we go from the major release `23.1.0` to:
* `23.1.0.1`/`23.1.0.2-SNAPSHOT` on the HF branch
* `23.1.1-A1`/`23.1.1-SNAPSHOT` on the SP branch
* `23.2.0-A1`/`23.2.0-SNAPSHOT` on master.

I was instead expecting:
* `23.1.1`/`23.1.2-SNAPSHOT` on the HF branch
* `23.2.0-A1`/`23.2.0-SNAPSHOT` on the SP branch
* `24.1.0-A1`/`24.1.0-SNAPSHOT` on master.

Or a two branch solution (since we don't expect to release `2024.1.0` until after `2023.3.0` iirc):
* `23.1.1`/`23.1.2-SNAPSHOT` on the HF branch
* `23.2.0-A1`/`23.2.0-SNAPSHOT` on master.

I didn't think we supported four part version numbers in the new versioning scheme?

Reply:
Those 3 commits is what we historically used to do after a release.
At the moment new script does almost exactly what you described in "two branch solution":
- `23.1.1`/`23.1.1-SNAPSHOT` on the HF branch (easy to change to `23.1.2-SNAPSHOT`)
- `23.2.0-A1`/`23.2.0-A1-SNAPSHOT` on master (easy to change to `23.2.0-A1-SNAPSHOT` or even possibly better to `23.2.0-A2-SNAPSHOT` ?)

I think that, with time, we should add optional SP branch creation when next version (passed as `-d` argument to the script) is a major version bump. That way new SP branch would have same values as master for above examples and master would get whatever was passed in the script `-d` argument.
Does that make more sense?

Reply:
That sounds really good to me.

On the topic of SNAPSHOTs then I think they are a "snapshot" of what the next release will be and so `23.1.2-SNAPSHOT` is more accurate than `23.1.1-SNAPSHOT`. You're probably right about `23.2.0-A2-SNAPSHOT` - the weekly release will soon overwrite whatever we put, but it's overwhelmingly likely to be an A release next.